docs/design.md

```diff
@@ -0,0 +1,101 @@
+# Enclave Confidential Containers (Enclave-CC) Design and Architecture
+
+## Motivation
+
+The Confidential Containers project provides a way for cloud native applications to choose
+Confidential Compute during deployment without modification to the containers. Confidential Compute
+comes in two main flavors: process and virtual machine based isolation. The term enclave is often
+associated with process-based isolation.
+
+Enclave-CC enables process-based isolation. Process-based isolation is beneficial by drawing the
+isolation boundary exactly around each container process. This reduces the Trusted Computing Base,
+or the amount of the system your application depends on for security.
+
+Consistent with the VM-based isolation design, the enclave-cc approach does not measure and attest
+to the workload (your application). Instead it uses generic enclaves to isolate your containers from
+the rest of the system. Measurement and attestation of the empty generic enclaves is used in
+building trust to deliver the application securely.
+
+## Goals
+
+### Keep the same user experience among Confidential Containers
+
+Several existing LibOS containerization solutions pay more attentions to seamlessly running
+application into the enclave in enclave-unaware manner. However, those solutions don't provide the
+image creation in enclave-unaware manner. Taking SGX LibOS solutions as examples, the image creator
+requires extra steps to wrap LibOS runtimes into container image. This is a big burden for the user
+experience when using process-based Confidential Containers. Especially each type of LibOS employs
+its unique build technology to create container images.
+
+Theoretically, LibOS runtimes are not parts of container workload, so the wrapping operation should
+be removed from image creation flow. Instead, LibOS is purely used as a runtime or loader to host
+the sensitive workloads in enclave. This approach targets a usage model where the decision to use
+an enclave is at deployment time. One could for instance grab any image out of a registry and
+target the deployment at an enclave.
```

Comment:
we could highlight here that we're targeting a use case where existing unmodified encrypted containers can be supported using process-based isolation:

```suggestion
an enclave is at deployment time. One could for instance grab any unmodified encrypted container image out of a registry and
target the deployment at an enclave in the same way it's done with VM-based isolation.
```